components/builder-api/src/server/resources/channels.rs

```diff
@@ -505,7 +505,13 @@ fn demote_package(req: HttpRequest,
                                                              channel: channel.clone() },
                                        &*conn).map_err(Error::DieselError)
     {
-        Ok(_) => {
+        Ok(0) => {
+            debug!("Requested package {} for target {} not present in channel {}",
+                   ident, target, channel);
+            return HttpResponse::new(StatusCode::NOT_FOUND);
```

Comment:
minor nit: The `return` isn't really needed here as it's implicit.

I like the solution here. Just curious, from eyeballing the [promote code path](https://github.com/habitat-sh/builder/blob/d093c118d440ec339439533f89b1156bc6347aaf/components/builder-db/src/models/channel.rs#L321) it appears that it's susceptible to the same issue perhaps?

Reply:
It's possible, I'd have to look at the return type of diesels `get_result`. Since we `?` that call to get the channel and package ids, I'd assume we'll early return an error if channel or ident+target doesn't exist, but it is worth a closer look.